src/Nomad.Sorter.Infrastructure/Cosmos/ChangeFeed/ChangeFeedProcessorService.cs

```diff
@@ -0,0 +1,120 @@
+    {
+        _logger.LogInformation("Starting change feed processor service");
+        
+        Database database = await _cosmosClient.CreateDatabaseIfNotExistsAsync(_repositoryOptions.Value.DatabaseId,
+            cancellationToken: stoppingToken);
+        
+        Container leaseContainer =
+            await database.CreateContainerIfNotExistsAsync("lease", "/id", cancellationToken: stoppingToken);
+
+        //TODO: For a library a setup to define which containers you want to listen from would be best.
+        var parcelsContainer = database.GetContainer("parcels");
+
+        var changeFeedProcessor = parcelsContainer
+            .GetChangeFeedProcessorBuilder<JObject>(AppConstants.Name, OnChangesDelegate)
+            .WithErrorNotification((token, exception) =>
+            {
+                _logger.LogError(exception, "Failed {Token}", token);
+                return Task.CompletedTask;
+            })
+            .WithLeaseContainer(leaseContainer)
+            .WithInstanceName(Environment.MachineName)
+            .Build();
+        
+        await changeFeedProcessor.StartAsync();
+        
+        _logger.LogInformation("Successfully Started change feed processor service");
+
+        stoppingToken.Register(() =>
+        {
+            _logger.LogInformation("Stopping change feed processor");
+            changeFeedProcessor.StopAsync().Wait(TimeSpan.FromSeconds(5));
+            _logger.LogInformation("Stopped change feed processor");
+        });
+    }
+
+    private async Task OnChangesDelegate(ChangeFeedProcessorContext context, IReadOnlyCollection<JObject> changes,
+        CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("Changes detected {ChangeCount}", changes.Count);
+        foreach (var change in changes)
+        {
+            var typeToken = change["type"];
+
+            if (typeToken is null)
+            {
+                _logger.LogInformation("Item does not have required type field {Json}", change.ToString());
+            }
+
+            var type = typeToken.Value<string>();
+
+            var task = type switch
+            {
+                nameof(Parcel) => InvokeHandler(typeof(Parcel), change, cancellationToken),
+                _ => NotSupported(type)
+            };
+
+            await task;
+        }
+    }
+
+    private async Task InvokeHandler(Type itemType, JObject instance, CancellationToken cancellationToken)
+    {
+        var item = instance.ToObject(itemType);
+
+        Type? handlerType = null;
+
+        if (Handlers.ContainsKey(itemType) is false)
+        {
+            handlerType = typeof(IChangeFeedItemProcessor<>).MakeGenericType(itemType);
+            Handlers[itemType] = handlerType;
+        }
+
+        handlerType ??= Handlers[itemType];
+        
+        await (ValueTask) handlerType.GetMethod("HandleAsync")?
+            .Invoke(_serviceProvider.GetRequiredService(handlerType), new[] {item, cancellationToken})!;
```

Comment:
The `?.` later followed by an `!` is a bit concerning. This will `throw` if it's `null`.

```suggestion
        await (ValueTask) handlerType.GetMethod("HandleAsync")?
            .Invoke(_serviceProvider.GetRequiredService(handlerType), new[] { item, cancellationToken })!;
```